Sources/SwiftDocC/Model/Rendering/RenderNodeTranslator.swift

```diff
@@ -1969,3 +1977,16 @@ extension TutorialArticleSection: RenderTree {}
 extension ContentLayout: RenderTree {}
 
 extension ContentRenderSection: RenderTree {}
+
+private extension Set where Element == SourceLanguage {
+    func matchesOneOf(traits: Set<DocumentationDataVariantsTrait>) -> Bool {
```

Comment:
I'm sure this will work great, especially since there are likely only 1-2 traits most of the time. But in general I wonder if it would be more efficient to try to use a set-intersection and not a repeated call to `contains`. So map the traits to languages ahead of time, creating a set of trait languages, and then use a single call to `Set#intersection` or `formIntersection` and then `isEmpty`.

Reply:
Reasoning about performance it always difficult without benchmarking but this question made me curious so I threw together a small micro benchmark using https://github.com/apple/swift-collections-benchmark with sets of variants traits and source languages with string values of 512 random letters (`[A-Za-z]{512}`).

I tested 6 different implementations:
- this
- `!isDisjoint(with: Set(traits.map { .init(name: $0.interfaceLanguage!) }))`
- `!isDisjoint(with: traits.map { .init(name: $0.interfaceLanguage!) })`
- `!isDisjoint(with: traits.lazy.map { .init(name: $0.interfaceLanguage!) })`
- `!intersection(Set(traits.map { .init(name: $0.interfaceLanguage!) })).isEmpty`
- `!intersection(traits.map { .init(name: $0.interfaceLanguage!) }).isEmpty`

When the variant traits and source languages had no overlap (achieved by lowercasing the variant traits and uppercasing the source languages) there was no difference between any of these implementations except that the two implementations that created an intermediate `Set` was slightly slower.
<img width="1280" alt="disjoint-inputs" src="https://github.com/apple/swift-docc/assets/1228143/e4f563a7-7c68-426b-97d7-7bf9a4f7ab38">

However, when the variant traits and source languages were created from the same string identifiers the `contains(where:)` implementation was the only one that didn't grow with its input size. 

<img width="1280" alt="intersecting-inputs" src="https://github.com/apple/swift-docc/assets/1228143/c3f8c555-2033-46c9-b83c-665c9a279075">

If I loop over the set of traits and make individual checks if the set of languages contains a set of only that single trait—which _isn't_ how the `RenderNodeTranslator` uses this check—then again there's no difference between the implementations except that the implementations that created an intermediate `Set` was slightly slower.
<img width="1280" alt="intersecting-individual-inputs" src="https://github.com/apple/swift-docc/assets/1228143/b81356ae-a532-44cc-87fa-641de3a89e4e">

Reply:
Wow these charts are amazing! Thanks for doing this. I'll have to try swift-collections-benchmark someday. The first thing that strikes me is that the vertical scales are microseconds at worst. And you even have nanoseconds and picoseconds :)  So clearly this code is fast no matter what we do.

I wonder why the `contains(where:)` version is faster in the second case? Is it as simple as always quickly finding the match on the first element? vs. processing all the elements?